app/get-existing-data-full.js

```diff
@@ -0,0 +1,17 @@
+const db = require('./data')
+
+const getExistingDataFull = async (correlationId, sourceSystem, frn, agreementNumber, transaction) => {
+  return db.reportData.findOne({
+    transaction,
+    where: {
+      correlationId,
+      sourceSystem,
+      frn,
+      agreementNumber
```

Comment:
Similarly to comment on get-delta-amount. we probably also want a similar filtering system here.